cmd/mps-control-daemon/mps/daemon.go

```diff
@@ -124,25 +127,51 @@ func (d *Daemon) Start() error {
 	}
 	defer statusFile.Close()
 
+	klog.InfoS("Starting log tailer", "resource", d.rm.Resource())
+	tail := exec.Command("tail", "-n", "+1", "-f", filepath.Join(logDir, "control.log")) //nolint:gosec
+	tail.Stdout = os.Stdout
+	if err := tail.Start(); err != nil {
+		klog.ErrorS(err, "Could not start tail command on control.log; ignoring logs")
+		d.tail = nil
+	} else {
+		d.tail = tail
+	}
+
 	return nil
 }
 
 // Stop ensures that the MPS daemon is quit.
 func (d *Daemon) Stop() error {
-	output, err := d.EchoPipeToControl("quit")
+	_, err := d.EchoPipeToControl("quit")
 	if err != nil {
 		return fmt.Errorf("error sending quit message: %w", err)
 	}
-	klog.InfoS("Shut down MPS", "output", output)
+	klog.InfoS("Stopped MPS control daemon", "resource", d.rm.Resource())
+
+	if d.tail != nil {
+		klog.InfoS("Stopping log tailer", "resource", d.rm.Resource())
+		if err := d.tail.Process.Signal(os.Kill); err != nil {
+			klog.ErrorS(err, "Failed to stop log tailer")
+		}
+
+		err := d.tail.Wait()
```

Comment:
Question -- what happens if an error is returned when sending the kill signal to the `tail` process? Should we skip this `Wait()` call in that case? I would expect the tail process to eventually get reaped anyways once this container exits.

Reply:
Yes, I was going to ask the same thing.

Reply:
Yes, that probably makes sense. I have updated the implementation to not call `Wait` in the event of an error. (and also switched to `SIGTERM`).

Reply:
See my comment [here](https://github.com/NVIDIA/k8s-device-plugin/pull/575#discussion_r1523406298) which avoids this entirely.